tachyon/crypto/sumcheck/multilinear/data_structures.h

```diff
@@ -0,0 +1,167 @@
+#ifndef TACHYON_CRYPTO_SUMCHECK_MULTILINEAR_DATA_STRUCTURES_H_
+#define TACHYON_CRYPTO_SUMCHECK_MULTILINEAR_DATA_STRUCTURES_H_
+
+#include <algorithm>
+#include <memory>
+#include <numeric>
+#include <vector>
+
+#include "absl/container/node_hash_map.h"
+
+#include "tachyon/base/buffer/copyable.h"
+#include "tachyon/base/containers/contains.h"
+#include "tachyon/base/logging.h"
+#include "tachyon/math/polynomials/multivariate/multilinear_extension.h"
+
+namespace tachyon {
+
+namespace crypto::sumcheck {
+
+// This data structure will be used as the verifier key.
+struct TACHYON_EXPORT PolynomialInfo {
+  size_t max_multiplicands;
+  size_t num_variables;
+
+  bool operator==(const PolynomialInfo& other) const {
+    return max_multiplicands == other.max_multiplicands &&
+           num_variables == other.num_variables;
+  }
+  bool operator!=(const PolynomialInfo& other) const {
+    return !operator==(other);
+  }
+  // TODO(ashjeong): For better unit testing
+  // PolynomialInfo Random() {}
+};
+
+template <typename Evaluations>
+class ListOfProductsOfPolynomials;
+
+template <typename Evaluations>
+class ListOfProductsOfPolynomials<math::MultilinearExtension<Evaluations>> {
+ public:
+  using F = typename math::MultilinearExtension<Evaluations>::Field;
+
+  struct Term {
+    F coefficient;
+    std::vector<size_t> indexes;
+  };
```

Comment:
I'm not sure this should be moved out, since the elements in the vector `indexes` rely on member `flattened_ml_extensions` within the `ListOfProductsOfPolynomials` class.

Reply:
Got it!